src/Authenticator/LoginHandler.php

```diff
@@ -491,6 +491,13 @@ public function redirectAfterSuccessfulLogin(): HTTPResponse
         }
         $request->getSession()->clear(static::SESSION_KEY . '.mustLogin');
 
+        // Deactivate sudo mode that was activated in doLogin()
+        $service = $this->getSudoModeService();
+        // Check if the service has a deactivate method, because it is not defined on the interface
+        if (method_exists($service, 'deactivate')) {
+            call_user_func([$service, 'deactivate'], $this->getRequest()->getSession());
```

Comment:
Why are we deactivating it, though? Shouldn't we just be not activating it in the first place?

Reply:
See https://github.com/silverstripe/silverstripe-framework/pull/11582#discussion_r1957424413

Reply:
Have moved to doPerformLogin()